src/model.ts

```diff
@@ -931,6 +932,75 @@ export class GitExtension implements IGitExtension {
     }
   }
 
+  /**
+   * Collects files that have changed on the remote branch.
+   *
+   */
+  async remoteChangedFiles(): Promise<Git.IStatusFile[]> {
+    // if a file is changed on remote add it to list of files with appropriate status.
+    this._remoteChangedFiles = [];
+    let remoteChangedFiles: null | string[] = null;
+    if (this.status.remote && this.status.behind > 0) {
+      remoteChangedFiles = (
+        await this._changedFiles('WORKING', this.status.remote)
+      ).files;
+      remoteChangedFiles?.forEach(element => {
+        this._remoteChangedFiles.push({
+          status: 'remote-changed',
+          type: this._resolveFileType(element),
+          x: '?',
+          y: 'B',
+          to: element,
+          from: '?',
+          is_binary: false
+        });
+      });
+    }
+    return this._remoteChangedFiles;
+  }
+
+  /**
+   * Notifies user is a file that is attached has is behind changes in the remote branch with a pop-up Dialog
+   *
+   */
+  async checkRemoteChangeNotified(): Promise<void> {
```

Comment:
Dialogs are really annoying by nature (they break the user flow). So they should be used sparingly. Therefore I suggest displaying at most one dialog listing all the files that are behind. And in the dialog I would add a button to allow the user to pull the changes directly.

Finally I would make this notification optional.